staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go

```diff
@@ -90,6 +99,14 @@ var _ recognizer.RecognizingDecoder = &Serializer{}
 
 const serializerIdentifier runtime.Identifier = "protobuf"
 
+func getSlice(size int) []byte {
+	in := pool.Get().([]byte)
+	if cap(in) < size {
+		return make([]byte, size, (size+1)*2)
```

Comment:
I will leave a comment on the other PR suggesting another approach to try.

But for this approach, you should bucket sizes e.g. by power of 2 and use separate pools for each size.